### PR TITLE
[release-ocm-2.11] NO-ISSUE: Fix subsystem tests in the CI - WireMock stubs script is failing to reach WireMock from time to time

### PR DESCRIPTION
**Pull request by danmanor**
tailored backport of - https://github.com/openshift/assisted-service/pull/6569

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by danmanor** [comment]
/override ci/prow/e2e-ai-operator-ztp ci/prow/e2e-ai-operator-ztp-sno-day2-workers ci/prow/e2e-ai-operator-ztp-sno-day2-workers-late-binding

File: tools/deploy_wiremock.py [hunk 1]
```diff
@@ -23,6 +34,86 @@ def deploy_wiremock(deploy_options):
...
+def deploy_ingress(hostname, deploy_options):
+    src_file = os.path.join(os.getcwd(), 'deploy', "wiremock", "wiremock-ingress.yaml")
+    dst_file = os.path.join(os.getcwd(), 'build', deploy_options.namespace, "wiremock-ingress.yaml")
+    with open(src_file, "r") as src:
+        with open(dst_file, "w+") as dst:
+            data = src.read()
+            data = data.replace('REPLACE_NAMESPACE', deploy_options.namespace)
+            data = data.replace("REPLACE_HOSTNAME", hostname)
+            log.info(f"Deploying {dst_file}")
+            dst.write(data)
```
Comment (eifrach) [inline tools/deploy_wiremock.py:70]: can we use here `yaml.load` and update the value as a dict.
and `yaml.dump` to write it to a file.

File: tools/deploy_wiremock.py [hunk 2]
```diff
@@ -23,6 +34,86 @@ def deploy_wiremock(deploy_options):
...
+        expected_exceptions=(RuntimeError),
+        sleep_seconds=SLEEP, waiting_for="Stubs to be populated"
+    )
+
+
+def is_wiremock_service_ready(namespace: str) -> bool:
+    return get_service_external_ip(namespace=namespace) != "null"
+
+
+def wait_for_wiremock_service(namespace: str):
```
Comment (eifrach) [inline tools/deploy_wiremock.py:97]:
```suggestion
def wait_for_wiremock_service(namespace: str) -> None:
```

File: tools/deploy_wiremock.py [hunk 3]
```diff
@@ -23,6 +34,86 @@ def deploy_wiremock(deploy_options):
...
+    waiting.wait(
+        lambda: is_wiremock_service_ready(namespace=namespace),
+        timeout_seconds=TIMEOUT,
+        expected_exceptions=(requests.exceptions.ConnectionError, requests.exceptions.ReadTimeout),
+        sleep_seconds=SLEEP, waiting_for="Wiremock service to be ready")
+
+
+def wait_for_wiremock_pod(url: str):
```
Comment (eifrach) [inline tools/deploy_wiremock.py:105]:
```suggestion
def wait_for_wiremock_pod(url: str) -> None:
```

**Review by eifrach — CHANGES_REQUESTED** [review]

File: tools/deploy_wiremock.py (hunk 1 above)
Comment (danmanor) [reply tools/deploy_wiremock.py:70]: I agree, I was just following the pattern in other scripts for consistency, also this is how the code in master already is

File: tools/deploy_wiremock.py (hunk 2 above)
Comment (danmanor) [reply tools/deploy_wiremock.py:97]: No need, this is actually how linters do

File: tools/deploy_wiremock.py (hunk 3 above)
Comment (danmanor) [reply tools/deploy_wiremock.py:105]: Same

Comment (eifrach) [reply tools/deploy_wiremock.py:105]: it's not for linters

**Pull request merged**